app/services/save_provider_user.rb

```diff
@@ -13,15 +13,21 @@ def call!
 private
 
   def update_permissions
+    unselected_permissions = ProviderPermissions
+      .where(provider_user: @provider_user)
+      .where.not(provider_id: @permissions.values.flatten)
+
     ActiveRecord::Base.transaction do
-      ProviderPermissions
-        .where(provider_user: @provider_user)
-        .update_all(ProviderPermissionsOptions.reset_attributes)
+      ProviderPermissionsOptions::VALID_PERMISSIONS.each do |permission|
+        unselected_permissions.where(permission => true).each { |perm| perm.update(permission => false) }
+      end
 
       @permissions.each do |permission, provider_ids|
-        ProviderPermissions
-          .where(provider_user: @provider_user, provider_id: provider_ids)
-          .update_all(permission => true)
+        provider_ids.each do |provider_id|
+          ProviderPermissions
+            .where(provider_user: @provider_user, provider_id: provider_id, permission => false)
+            .update(permission => true)
```

Comment:
```suggestion
            .update!(permission => true)
```

Reply:
Ah yes, I was wondering why that wouldn't work for `ActiveRecord::Relation` wasn't paying attention to the error - updated here https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/1905/commits/20570652057d73a36bb057f241de4ba031e9e40a